src/Content/OEmbed.php

```diff
@@ -131,14 +131,12 @@ public static function fetchURL($embedurl, $no_rich_type = false)
 		$j->embedurl = $embedurl;
 
 		// If fetching information doesn't work, then improve via internal functions
-		if (($j->type == "error") || ($no_rich_type && ($j->type == "rich"))) {
+		if ($no_rich_type && ($j->type == "rich")) {
```

Comment:
Is the error case now handled differently?

Reply:
What was happening was the following:
- We tried OEmbed or we skipped to custom attachment display number 1
- If OEmbed failed, we still provided some sort of data structure, which lead to custom attachment display number 2

Now it's more streamlined, if OEmbed data can't be found we just revert to custom attachment display number 1 no matter what which gives more consistent results across the board. The condition still makes sense for the rich type so I kept it.

Reply:
I see a problem there, since not every page is returning OEmbed content at all. And AFAIK the returned OEmbed data could be "rich" but could be some link data as well - the same that is returned with the additional function.

So when we are now querying a page, we won't get any information at all when they don't support OEmbed.

I have the feeling that we will have much fun until with this we can do the release. For example: Currently I'm having problems with just adding some links without description. Possibly we should rework the whole strategy what to query and when.

Reply:
I agree about the rework, it shouldn’t be up to OEmbed to query OpenGraph data for example.

What I would like is a succession of fallbacks starting with OEmbed, then OpenGraph, then the [attachment] tag attributes.

Reply:
BTW: When you look into the OpenGraph code you will see that OEmbed is queried as well :-)

And yeah: We really have to clean up this mess. I would like to do it in some way so that we rely on ```[attachment]``` whenever this is present and to do this OEmbed stuff only video tags and OpenGraph/Oembed during import from system that do not append preview data at all.

So the only question now is: Have you checked with deactivated OEmbed and on pages without OEmbed to check if there is still output?

Currently there seems to be issues that empty data is generated with some links.

Reply:
I didn't check without OEmbed enabled because the problem was in the OEmbed functions. I'll check tonight.

Reply:
Thanks!